src/OpenLoco/src/GameCommands/VehicleSpeedControl.cpp

```diff
@@ -0,0 +1,52 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Ui/WindowManager.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004BAB63
+    static uint32_t vehicleSpeedControl(const VehicleSpeedControlArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        auto* head = EntityManager::get<Vehicles::VehicleHead>(args.head);
+        if (head == nullptr)
+        {
+            return FAILURE;
+        }
+
+        head->var_6E = args.speed;
```

Comment:
is it possible to name `var_6E` now?

Reply:
Yeah, it's a manual target speed of some point. Duncan already indicated as much in a comment in the vehicle struct as well.